cuda_core/cuda/core/experimental/_stream.py

```diff
@@ -43,6 +43,47 @@ class StreamOptions:
     priority: Optional[int] = None
 
 
+class IsStreamT(Protocol):
+    def __cuda_stream__(self) -> Tuple[int, int]:
+        """
+        For any Python objects that are meant to be interpreted as a CUDA stream, they
+        can do so by implementing this protocol that returns a 2-tuple: The protocol
+        version number (currently ``0``) and the address of ``cudaStream_t``. Both values
+        should be Python `int`.
+        """
+        ...
+
+
+def _try_to_get_stream_ptr(obj: IsStreamT):
+    try:
+        cuda_stream_attr = obj.__cuda_stream__
+    except AttributeError:
+        raise TypeError(f"{type(obj)} object does not have a '__cuda_stream__' attribute") from None
+
+    if callable(cuda_stream_attr):
+        info = cuda_stream_attr()
+    else:
+        info = cuda_stream_attr
+        warnings.simplefilter("once", DeprecationWarning)
+        warnings.warn(
+            "Implementing __cuda_stream__ as an attribute is deprecated; it must be implemented as a method",
+            stacklevel=3,
+            category=DeprecationWarning,
+        )
+
+    try:
+        len_info = len(info)
+    except Exception as e:
```

Comment:
Is there any way we can more tightly scope the exception we catch here?

Reply:
This was from the old code refactored to here. Narrowed it down to `TypeError` in commit f65b644a.